File: backend/onGo-domain/src/main/kotlin/com/ongo/domain/fanreward/FanReward.kt
package com.ongo.domain.fanreward

import java.time.LocalDateTime

data class FanReward(
    val id: Long? = null,
    val userId: Long,
    val name: String,
    val description: String? = null,
    val pointsCost: Int = 0,
    val category: String = "BADGE",
    val isActive: Boolean = true,
    val claimedCount: Int = 0,
    val maxClaims: Int? = null,
    val imageUrl: String? = null,
    val createdAt: LocalDateTime? = null,
)

File: frontend/src/types/fanReward.ts
export interface FanReward {
  id: number
  name: string
  description: string
  pointsCost: number
  category: 'BADGE' | 'EXCLUSIVE' | 'SHOUTOUT' | 'MERCH' | 'MEETUP'
  isActive: boolean
  claimedCount: number
  maxClaims: number | null
  imageUrl: string | null
  createdAt: string
}

export interface FanActivity {
  id: number
  fanName: string
  activityType: 'COMMENT' | 'LIKE' | 'SHARE' | 'WATCH' | 'SUBSCRIBE'
  points: number
  description: string
  timestamp: string
}

export interface FanRewardSummary {
  totalRewards: number
  activeRewards: number
  totalPointsDistributed: number
  totalClaims: number
  topCategory: string
}
